### DISCUSSION TITLE
I add a metric to /config/oal/my.oal but see nothing

### DISCUSSION THREAD

**Discussion by xing393939** [apache/skywalking#8748] [Q&A] [answered]
My sw is 8.9.1 and I add a metric to /config/oal/my.oal like this:
`my_endpoint_cpm = from(Endpoint.*).cpm();`
Then, reboot the OAP server, but when I query data use graphql I get nothing, I also do not see the index in elasticsearch. 

btw, I query ok when use endpoint_cpm metric:
![im](http://r92kkjyoq.hn-bkt.clouddn.com/1.jpg)

**Comment by wu-sheng** [comment]
In current ElasticSearchstorage, there is no point to create new index. This metric is a new metric in the cpm index.
Also, your GraphQL is not using your new metric name, what do you want to see?

**Reply by xing393939** [reply]
Thanks, now I can see the data in sw_metrics-cpm-xxx, but I wonder how to use it in skywalking rocketbot ui, I can not replace endpoint_cpm with my_endpoint_cpm as my picture point at.
![img](http://r92kkjyoq.hn-bkt.clouddn.com/2.jpg)

**Reply by wu-sheng** [reply]
What do you mean you can't? You need to enter after changing metric name.

**Reply by xing393939** [reply]
I am sorry I did not explan it cleanly. Of caurse I enter the new name `my_endpoint_cpm`, but after I enter the name, the select field that my picture point out is empty and has no options. Then I saved the setting and the chart is show nothing.
![im](http://r92kkjyoq.hn-bkt.clouddn.com/3.jpg)

**Reply by wu-sheng** [reply]
This seems like the backend doesn't have this metric. I doubt you could find this metric in `sw_metrics-cpm-xxx`. That index would includes all metrics using `cpm` function, you have to find `metric-name= my_endpoint_cpm`.

From my understanding, either your OAL script is never loaded, or there is some booting stage error, but you ignored.

**Reply by xing393939** [reply]
Finally, I put the statement in the config/oal/core.oal and it works. 
This days I put the statement in the config/oal/my.oal which created by myself, and it did not work!

**Reply by wu-sheng** [reply]
Oh, you can't add a new OAL script w/o code changes. They are read through codes.

**Comment by wu-sheng** [comment, marked as answer]
> Finally, I put the statement in the config/oal/core.oal and it works. This days I put the statement in the config/oal/my.oal which created by myself, and it did not work!

Conclusion ⬆️
